greenlight/lib/tests/test_tabs.py

```diff
@@ -47,9 +47,7 @@ def tearDown(self):
             gBrowser.removeAllTabsBut(gBrowser.tabs[0]);
         """)
 
-        # Needs to be set by the harness itself
-        self.prefs.restore_pref('browser.tabs.warnOnClose')
-        self.prefs.restore_pref('browser.tabs.warnOnCloseOtherTabs')
+        self.wait_for_condition(lambda _: len(self.browser.tabbar.tabs) == 1)
```

Comment:
When I look at http://mxr.mozilla.org/mozilla-central/source/browser/base/content/tabbrowser.xml#1938 I don't see that it is asynchronous, but this check doesn't hurt. So all fine here! Thanks for the explanation.